validator/coordinator/src/main/java/tech/pegasys/artemis/validator/coordinator/ValidatorCoordinator.java

```diff
@@ -101,150 +116,173 @@ public void onNewHeadStateEvent(HeadStateEvent headStateEvent) {
+    UnsignedLong epoch = BeaconStateUtil.slot_to_epoch(slot);
+
+    Bytes32 messageHash =
+        HashTreeUtil.hash_tree_root(SSZTypes.BASIC, SSZ.encodeUInt64(epoch.longValue()));
+    int domain =
+        BeaconStateUtil.get_domain(state.getFork(), epoch, Constants.DOMAIN_RANDAO).intValue();
+    return getSignature(messageHash, domain, proposer);
+  }
+
+  private BLSSignature getBlockSignature(
+      BeaconState state, BeaconBlock block, BLSPublicKey proposer) {
+    int domain =
+        BeaconStateUtil.get_domain(
+                state.getFork(),
+                BeaconStateUtil.slot_to_epoch(UnsignedLong.valueOf(block.getSlot())),
+                Constants.DOMAIN_BEACON_BLOCK)
+            .intValue();
+
+    Bytes32 blockRoot = block.signed_root("signature");
+
+    return getSignature(blockRoot, domain, proposer);
+  }
+
+  private BeaconBlock createInitialBlock(BeaconStateWithCache state, BeaconBlock oldBlock) {
+    Bytes32 blockRoot = oldBlock.signed_root("signature");
+    List<Attestation> current_attestations;
+    final Bytes32 MockStateRoot = Bytes32.ZERO;
+    BeaconBlock newBlock;
+    if (state
+            .getSlot()
+            .compareTo(
+                UnsignedLong.valueOf(
+                    Constants.GENESIS_SLOT + Constants.MIN_ATTESTATION_INCLUSION_DELAY))
+        >= 0) {
+      UnsignedLong attestation_slot =
+          state.getSlot().minus(UnsignedLong.valueOf(Constants.MIN_ATTESTATION_INCLUSION_DELAY));
+
+      current_attestations = this.store.getUnprocessedAttestationsUntilSlot(attestation_slot);
+
+      newBlock =
+          DataStructureUtil.newBeaconBlock(
+              state.getSlot().plus(UnsignedLong.ONE),
+              blockRoot,
+              MockStateRoot,
+              newDeposits,
+              current_attestations);
+    } else {
+      newBlock =
+          DataStructureUtil.newBeaconBlock(
+              state.getSlot().plus(UnsignedLong.ONE),
+              blockRoot,
+              MockStateRoot,
+              newDeposits,
+              new ArrayList<>());
+    }
+    return newBlock;
+  }
+
   private void initializeValidators() {
     // Add all validators to validatorSet hashMap
     int nodeCounter = UInt256.fromBytes(nodeIdentity.bytes()).mod(numNodes).intValue();
-    // LOG.log(Level.DEBUG, "nodeCounter: " + nodeCounter);
-    // if (nodeCounter == 0) {
 
     int startIndex = nodeCounter * (numValidators / numNodes);
     int endIndex =
         startIndex
             + (numValidators / numNodes - 1)
             + toIntExact(Math.round((double) nodeCounter / Math.max(1, numNodes - 1)));
     endIndex = Math.min(endIndex, numValidators - 1);
-    // int startIndex = 0;
-    // int endIndex = numValidators-1;
     LOG.log(Level.DEBUG, "startIndex: " + startIndex + " endIndex: " + endIndex);
     for (int i = startIndex; i <= endIndex; i++) {
       BLSKeyPair keypair = BLSKeyPair.random(i);
+      int port = 50000 + i;
```

Comment:
make a todo to externalize port 5000 as the starting point.  or make it a constant